lib/eex/lib/eex/engine.ex

```diff
@@ -98,21 +101,24 @@ defmodule EEx.Engine do
   @doc """
   Returns an empty string as initial buffer.
   """
-  def init(_opts) do
+  @spec init(Keyword.t) :: <<>>
+  def init(opts) when is_list(opts) do
```

Comment:
Plus this is not required to return a string. The buffer can be anything. Since those are callback implementation and the callback already specifies the types, I would remove those specs. Same for `handle_*` below.